### PR TITLE
Memphis week4 ass5

### PR DESCRIPTION
**Pull request by memphis88**
If you have the time for a brief review you are welcome to go for it. If not, let me know so I can move to the next assignment with this code

### CODE REVIEW

File: memphis88/week4/ass5/app/mixins/on-pressing-enter-mixin.jsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+var OnPressingEnterMixin = {
```
Comment (pcolliander) [inline memphis88/week4/ass5/app/mixins/on-pressing-enter-mixin.jsx]: I think it seems overkill to have a mixin for this, you could just write one line to check for the keycode of Enter.

File: memphis88/week4/ass5/app/stores/message-store.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		case Constants.DELETE_MESSAGE:
+			var msgToDelete = _messages.find(function(l) { return l.id == payload.id });
+			_messages.splice(_messages.indexOf(msgToDelete), 1);
+			break;
+		case Constants.MOVE_MESSAGE:
+			var msgToMove = _messages.find(function(l) { return l.id == payload.id });
+			msgToMove.listKey = payload.listKey;
+			break;
+		case Constants.ARCHIVE_MESSAGE:
+			Dispatcher.waitFor([ArchiveMessageStore.dispatchToken]);
```
Comment (pcolliander) [inline memphis88/week4/ass5/app/stores/message-store.js:48]: The choice of separating messages from archived messages in this way seems to require a lot of code. First you have to maintain two separate stores with a `waitFor()` dependency between them and then you need separate views to handle them. I understand the idea but I think it's way easier just to add a `isArchived` attribute to each message, keep them all in your message store, and then filter them out based on that attribute instead.

File: memphis88/week4/ass5/app/components/archive-list.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		ArchiveListStore.addChangeListener(this._onChange);
+	},
+
+	componentWillUnmount: function() {
+		ArchiveListStore.removeChangeListener(this._onChange);
+	},
+
+	_onChange: function() {
+		this.setState({
+			messages: ArchiveListStore.getAllForList(this.props.myListKey)
```
Comment (pcolliander) [inline memphis88/week4/ass5/app/components/archive-list.jsx:29]: nice way of getting only the data you need :+1:

File: memphis88/week4/ass5/app/components/assignment-app.jsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+	propTypes: {
+		generatedLists: React.PropTypes.array,
+		listName: React.PropTypes.string,
+	},
+
+	getInitialState: function() {
+		return {
+			generatedLists: ListStore.getAll(),
+			listName: ''
```
Comment (pcolliander) [inline memphis88/week4/ass5/app/components/assignment-app.jsx]: You could put a [ref](https://facebook.github.io/react/docs/more-about-refs.html) on your input field instead. Get its value by `this.refs.inputField.value;` Clear it out with `this.refs.inputField.value = "";` Then you don't have to maintain state and make the view more stateless.

File: memphis88/week4/ass5/app/components/generated-list.jsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	],
+
+	propTypes: {
+		messageList: React.PropTypes.array,
+		text: React.PropTypes.string,
+	},
+
+	getInitialState: function () {
+		return {
+			text: ''
```
Comment (pcolliander) [inline memphis88/week4/ass5/app/components/generated-list.jsx]: same as above

**Comment by pcolliander** [comment]
All in all I think it was pretty straightforward code. But I think you could make it easier for yourself if you made "archived" an attribute of each message. But that's just another way of doing it I suppose. Then you wouldn't have to hack around the circular dependency between the stores either.

**Comment by hulvei3** [comment]
@memphis88 @pcolliander fell free to merge it yourself when ready

**Comment by memphis88** [comment]
Keeping a separate model entity for archived messages is just me being a bit _stubborn_ on wanting to have distinct entities with different semantics (archived messages cannot be deleted, moved etc). I realize that since we are in the representational layer, it does not matter that much to keep a model strictly coupled to the _business logic_ but I wanted to experiment a bit with inter-store communication and `waitFor()`. Truth is that this complicates things even more in the next assignment, I'll consider changing the model if it gets unnecessarily complicated.

**Comment by hulvei3** [comment]
Good reasoning and reflection. I like that - regardless of the result.

**Pull request merged**